chart/etcd-backup-restore/templates/etcd-statefulset.yaml

```
@@ -165,6 +165,8 @@ spec:
         - name: STORAGE_CONTAINER
           value: {{ .Values.backup.storageContainer }}
 {{- if eq .Values.backup.storageProvider "S3" }}
+        - name: "AWS_SSE_CUSTOMER_KEY"
+          value: {{ .Values.sseCustomerKey }}
```

This is an optional field for S3 in the Helm template. There should be `if` construct guarding this template field.

```suggestion
        {{- if .Values.sseCustomerKey }}
        - name: "AWS_SSE_CUSTOMER_KEY"
          value: {{ .Values.sseCustomerKey }}
        {{- end }}
```

If there's a better way to achieve this, please follow that. But there has to be a conditional around SSE template value in this chart to ensure that this field is not *expected*.

Just another heads up - the helm charts for etcd-backup-restore are not *fully* up to date since the Gardener organization makes use of the etcd operator [etcd-druid](https://github.com/gardener/etcd-druid) to deploy etcd-backup-restore as a side-car container to an etcd running in [`etcd-wrapper`](https://github.com/gardener/etcd-wrapper).

Please validate that the charts are working for *you* as expected for etcd-backup-restore.

---

I've now added the `if` statement to make it optional. And yes I did see an open issue about the helm charts. I'll take a look to make sure it is working for us